### ISSUE TITLE
Docker build fails due to Cython not being found

### ISSUE THREAD

**Issue by DonMaEn** [SmartForest-no/SegmentAnyTree#1]
Hello!

Thank you for releasing this project and including a dockerfile with it! This is really helpful for projects like this with complicated dependencies.

I am trying to build a container from the provided Dockerfile, however I am running into issues. I am actually using podman rather than Docker, and I am on a redhat 8 system, but based on the error I am seeing I don't think that is the issue. I just wanted to note it.

The only command I ran to build the container was I ran the `build.sh` script from the repo.

The error I am getting is as follows:
`
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
torch-points-kernels 0.7.0 requires numpy<1.20, but you have numpy 1.24.4 which is incompatible.
Successfully installed numpy-1.24.4
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
WARNING: You are using pip version 21.2.4; however, version 24.2 is available.
You should consider upgrading via the '/usr/bin/python3.8 -m pip install --upgrade pip' command.
setup.py:10: UserWarning: No module named 'Cython'
  warnings.warn(e.args[0])
setup.py:94: UserWarning: Due to incompatibilities with Python 3.7 hdbscan nowrequires Cython to be installed in order to build it
  warnings.warn('Due to incompatibilities with Python 3.7 hdbscan now'
Traceback (most recent call last):
  File "setup.py", line 96, in <module>
    raise ImportError('Cython not found! Please install cython and try again')
ImportError: Cython not found! Please install cython and try again
Error: building at STEP "RUN wget https://github.com/scikit-learn-contrib/hdbscan/archive/master.zip &&     unzip master.zip &&     rm master.zip &&     cd hdbscan-master &&     python3.8 -m pip install -r requirements.txt &&     python3.8 setup.py install": while running runtime: exit status 1
`

It seems like the Dockerfile is pulling directly from a github repository without specifying a specific version of the repo it is pulling from. Is it possible this repo was updated recently and that is causing this issue? If not, do you have any other ideas about what might be causing this?

Thanks again!

**Comment by DonMaEn** [comment]
I managed to fix this, but I can't push to a new branch because I don't have permission. This is the git diff of the Dockerfile changes I made:
```
git diff Dockerfile
diff --git a/Dockerfile b/Dockerfile
index daebb78..fd10166 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -217,10 +217,12 @@ RUN TORCH_CUDA_ARCH_LIST=${TORCH_CUDA_ARCH_LIST_VER} FORCE_CUDA=1 python3.8 -m p


 # Download, unzip, and install hdbscan
-RUN wget https://github.com/scikit-learn-contrib/hdbscan/archive/master.zip && \
-    unzip master.zip && \
-    rm master.zip && \
-    cd hdbscan-master && \
+RUN python3.8 -m pip install cython==0.29.37
+
+RUN wget https://github.com/scikit-learn-contrib/hdbscan/archive/refs/tags/0.8.29.zip && \
+    unzip 0.8.29.zip && \
+    rm 0.8.29.zip && \
+    cd hdbscan-0.8.29 && \
     python3.8 -m pip install -r requirements.txt && \
     python3.8 setup.py install
```

The issue was that HDBScan had a new release because of some changes to Cython. Using a set version of HDBScan and Cython fixed the issue. I haven't actually tested the image yet, but it builds now.

**Comment by maciekwielgosz** [comment]
@DonMaEn can you issue a pull request I will merge this ? Thanks !

**Comment by DonMaEn** [comment]
@maciekwielgosz I would create a pull request, but I cannot create a branch to make the pull request from because I don't have permission. Is there another way to create a pull request without creating a branch?

**Comment by maciekwielgosz** [comment]
Hello @DonMaEn,

Thank you for identifying and fixing the dependency conflict. I have updated all manually in the Docker file .

Regarding your inability to create a branch for a pull request, here’s a solution:

Fork the Repository: Go to the main repository page and click on the "Fork" button. This will create a copy of the repository under your GitHub account.
Create a Branch in Your Forked Repository: In your fork, create a branch for the changes. For instance, you could call it fix/hdbscan-cython-dependency.
Commit Your Changes: Add your modifications to this branch in your fork.
Submit a Pull Request: Once your branch is ready, navigate to the original repository and click on “Pull requests,” then “New pull request.” From there, you can choose to make a pull request from your branch in your forked repository to the original repository.
This process doesn’t require permission to create a branch in the main repo, but it allows you to make a pull request that @maciekwielgosz or other maintainers can review and merge.
